File: LeetCode/1423. Maximum Points You Can Obtain from Cards/solution.cpp
/**
 * 40 / 40 test cases passed.
 * Status: Accepted
 * Runtime: 76 ms
 */
class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        int maxSum = accumulate(cardPoints.begin(), cardPoints.begin() + k, 0);
        int ans = maxSum;
        for (int i = 0; i < k; ++ i) {
            maxSum = maxSum - cardPoints[k - 1 - i] + cardPoints[cardPoints.size() - 1 - i];
            ans = max(ans, maxSum);
        }
        return ans;
    }
};

File: LeetCode/1423. Maximum Points You Can Obtain from Cards/solution.py
"""
40 / 40 test cases passed.
Status: Accepted
Runtime: 84 ms
"""
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        range_sum = sum(cardPoints[:k])
        ans = range_sum
        for i, j in zip(cardPoints[k-1::-1], cardPoints[:-k-1:-1]):
            range_sum = range_sum -i + j
            ans = max(ans, range_sum)
        return ans
